core/templates/domain/story/StoryObjectFactory.ts

```diff
@@ -43,27 +43,24 @@ export interface StoryBackendDict {
 }
 
 export class Story {
-  // Assigning union type of "string | null" to some fields,
-  // Because we are assigning null to those fields when we create
-  // interstitial story.
-  _id: string | null;
+  _id: string;
   _title: string;
   _description: string;
   _notes: string;
-  _storyContents: StoryContents | null;
+  _storyContents: StoryContents;
   _languageCode: string;
   _version: number;
-  _correspondingTopicId: string | null;
+  _correspondingTopicId: string;
   _thumbnailFilename: string | null;
   _thumbnailBgColor: string | null;
-  _urlFragment: string | null;
+  _urlFragment: string;
   _metaTagContent: string;
   constructor(
-      id: string | null, title: string, description: string,
-      notes: string, storyContents: StoryContents | null, languageCode: string,
-      version: number, correspondingTopicId: string | null,
+      id: string, title: string, description: string,
+      notes: string, storyContents: StoryContents, languageCode: string,
+      version: number, correspondingTopicId: string,
       thumbnailBgColor: string | null, thumbnailFilename: string | null,
```

Comment:
Both thumbnailBgColor and thumbnailFilename can be null because in the backend it is of type string and none. Take a look [here](https://github.com/oppia/oppia/blob/7aea401b372b38a66442e447332e4e06ca5880f6/core/domain/story_domain.py#L193).